pkg/controller/dependency/install.go

```diff
@@ -49,42 +50,50 @@ func init() {
 }
 
 // Setup vela dependency
-func Install(client client.Client) error {
-	log := ctrl.Log.WithName("vela dependency manager")
-	// Fetch the vela configuration
-	velaConfigNN := k8stypes.NamespacedName{Name: VelaConfigName, Namespace: types.DefaultOAMNS}
-	velaConfig := v1.ConfigMap{}
-	if err := client.Get(context.TODO(), velaConfigNN, &velaConfig); err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil
-		}
+func Install(kubecli client.Client) error {
+	velaConfig, err := fetchVelaConfig(kubecli)
+	if err != nil {
 		return err
 	}
-	for crd, chart := range velaConfig.Data {
-		log.Info("check on dependency", "crd resource", crd)
-		if err := client.Get(context.TODO(), k8stypes.NamespacedName{Name: crd}, &crdv1.CustomResourceDefinition{}); err != nil {
-			if apierrors.IsNotFound(err) {
-				if instErr := installHelmChart(client, []byte(chart), log); instErr != nil {
-					return errors.Wrap(instErr, "failed to install helm chart")
-				}
-			} else {
-				return err
-			}
-		} else {
-			log.Info("resources already exists, skip install", "crd", crd)
+	for _, chart := range velaConfig.Data {
+		err := installHelmChart(kubecli, []byte(chart), log)
```

Comment:
I don't think it's good to directly install without any check. We have a conclusion that we should finally use CRD Registry to install controllers. So we don't install controller if CRD exists.